packages/uikit-workshop/src/scripts/lit-components/pl-viewport-size/pl-viewport-size.js

```diff
@@ -52,12 +60,29 @@ class ViewportSize extends BaseLitComponent {
     }
   }
 
+  handlePixelUpdate(e) {
+    // Prevent inserting letters or symbols
+    if (e.key.match(/\D+/g) && !(e.charCode === 13 || e.keyCode === 13)) {
+      e.preventDefault();
+    } else {
+      this.setState({ inputPixelValue: e.target.value.replace(/\D+/g, '') });
+    }
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    this.iframe.sizeiframe(this.state.inputPixelValue, true);
+  }
+
   render() {
     if (!window.__PRERENDER_INJECTED) {
       return html`
-        <form class="pl-c-viewport-size" method="post" action="#">
-          <span class="pl-c-viewport-size__input">${this.px}px</span
-          >&nbsp;/&nbsp;
+        <form class="pl-c-viewport-size" @submit=${this.handleSubmit}>
+          <input
+            class="pl-c-viewport-size__input pl-c-viewport-size__input-action"
+            .value="${this.state.inputPixelValue}"
+            @keypress=${this.handlePixelUpdate}
```

Comment:
Also, I need to mention that I'm not that well known with React. Maybe there is a better approach for handling an input and replace/prevent an invalid key. In Angular, there are directives for that.